cmd/main.go

```diff
@@ -118,10 +118,18 @@ func main() {
 		cancel()
 	}()
 
+	apiRouter := routers.NewDefaultAPIRouter(optlyCache, conf.API, agentMetricsRegistry)
+	adminRouter := routers.NewAdminRouter(*conf)
+
+	if apiRouter == nil || adminRouter == nil {
+		log.Error().Msg("failed to start router, exiting")
+		cancel()
+	}
```

Comment:
We're loosing the granularity about which router is failing. We probably don't need to assign explicit variables since we can do the `nil` check within `NewServer` which should log with all the important details.

Reply:
This comment has not been addressed.